freelist.go

```diff
@@ -132,9 +132,9 @@ func (f *freelist) free(txid txid, p *page) {
 	allocTxid, ok := f.allocs[p.id]
 	if ok {
 		delete(f.allocs, p.id)
-	} else if (p.flags & (freelistPageFlag | metaPageFlag)) != 0 {
-		// Safe to claim txid as allocating since these types are private to txid.
-		allocTxid = txid
+	} else if (p.flags & freelistPageFlag) != 0 {
+		// Freelist is always allocated by prior tx.
+		allocTxid = txid - 1
```

Comment:
It can fix the issue, but it's a little complicated. No need to handle freelist as a special case. I think it can as simple as below. 


```
$ git diff
diff --git a/freelist.go b/freelist.go
index fdf8a36..4a15832 100644
--- a/freelist.go
+++ b/freelist.go
@@ -118,9 +118,6 @@ func (f *freelist) free(txid common.Txid, p *common.Page) {
        allocTxid, ok := f.allocs[p.Id()]
        if ok {
                delete(f.allocs, p.Id())
-       } else if p.IsFreelistPage() {
-               // Freelist is always allocated by prior tx.
-               allocTxid = txid - 1
        }
 
        for id := p.Id(); id <= p.Id()+common.Pgid(p.Overflow()); id++ {
```